packages/root-cms/ui/components/RichTextEditor/lexical/LexicalEditor.tsx

```diff
@@ -0,0 +1,184 @@
+import {HeadingNode} from '@lexical/rich-text';
+import {useMemo, useState} from 'preact/hooks';
+import {RichTextData} from '../../../../shared/richtext.js';
+import {joinClassNames} from '../../../utils/classes.js';
+import {
+  SharedHistoryProvider,
+  useSharedHistory,
+} from './hooks/useSharedHistory.js';
+import {ToolbarProvider} from './hooks/useToolbar.js';
+import {LexicalTheme} from './LexicalTheme.js';
+import {FloatingLinkEditorPlugin} from './plugins/FloatingLinkEditorPlugin.js';
+import {FloatingToolbarPlugin} from './plugins/FloatingToolbarPlugin.js';
+import {MarkdownTransformPlugin} from './plugins/MarkdownTransformPlugin.js';
+import {OnChangePlugin} from './plugins/OnChangePlugin.js';
+import {ShortcutsPlugin} from './plugins/ShortcutsPlugin.js';
+import {ToolbarPlugin} from './plugins/ToolbarPlugin.js';
+
+const INITIAL_CONFIG: InitialConfigType = {
+  namespace: 'RootCMS',
+  theme: LexicalTheme,
+  nodes: [
+    AutoLinkNode,
+    HeadingNode,
+    // ImageNode,
+    LinkNode,
+    ListNode,
+    ListItemNode,
+    // YouTubeNode,
+  ],
+  onError: (err: Error) => {
+    console.error('[LexicalEditor] error:', err);
+    throw err;
+  },
+};
+
+export interface LexicalEditorProps {
+  className?: string;
+  placeholder?: string;
+  value?: RichTextData | null;
+  onChange?: (value: RichTextData | null) => void;
+}
+
+export function LexicalEditor(props: LexicalEditorProps) {
+  // This component sets up the context providers and shell for lexical, and
+  // then renders the <Editor> component which can use the shared context states
+  // to render the rich text editor.
+  return (
+    <LexicalComposer initialConfig={INITIAL_CONFIG}>
+      <SharedHistoryProvider>
+        <ToolbarProvider>
+          <div className={joinClassNames(props.className, 'LexicalEditor')}>
+            <Editor
+              placeholder={props.placeholder}
+              value={props.value}
+              onChange={props.onChange}
+            />
+          </div>
+        </ToolbarProvider>
+      </SharedHistoryProvider>
+    </LexicalComposer>
+  );
+}
+
+interface EditorProps {
+  placeholder?: string;
+  value?: RichTextData | null;
+  onChange?: (value: RichTextData | null) => void;
+}
+
+function Editor(props: EditorProps) {
+  const {historyState} = useSharedHistory();
+  const [editor] = useLexicalComposerContext();
+  const [activeEditor, setActiveEditor] = useState(editor);
+  const [isLinkEditMode, setIsLinkEditMode] = useState(false);
+  const [floatingAnchorElem, setFloatingAnchorElem] =
+    useState<HTMLElement | null>(null);
+
+  // The "onRef" var is used as a callback, so it's typed to `any` here to avoid
+  // type warnings.
+  const onRef: any = (el: HTMLDivElement) => {
+    if (el) {
+      setFloatingAnchorElem(el);
+      // Prevent mouse clicks on link nodes from redirecting the browser tab.
+      el.addEventListener('click', (event) => {
```

Comment:
Google AI mode suggests this is another way to handle this (untested, not sure if it works)

```
import { $isLinkNode } from '@lexical/link';
import { $getSelection, $isRangeSelection } from 'lexical';

editor.registerCommand(
  CLICK_COMMAND,
  (payload) => {
    const event = payload;
    const selection = $getSelection();

    if ($isRangeSelection(selection)) {
      const node = selection.getNodes()[0]; // Or iterate through nodes if multiple are selected
      if ($isLinkNode(node)) {
        event.preventDefault(); // Prevent default link behavior
        return true; // Mark as handled
      }
    }
    return false; // Not handled
  },
  COMMAND_PRIORITY_LOW,
);
```

Reply:
done, realized similar logic already existed in `FloatingLinkEditorPlugin.tsx` and so i moved this logic there.